Plugin.php

```diff
@@ -91,6 +93,31 @@ public function boot()
                 ]
             ]);
         });
+
+        Event::listen('pages.menuitem.listTypes', function() {
+            return [
+                'forum-channel'=>'Forum channel',
+                'all-forum-channels'=>'All forum channels',
+                'forum-topic'=>'Forum topic',
+                'all-forum-topics'=>'All forum topics',
+            ];
+        });
+
+        Event::listen('pages.menuitem.getTypeInfo', function($type) {
+            if ($type == 'forum-channel' || $type == 'all-forum-channels') {
+                return Channel::getMenuTypeInfo($type);
+            } elseif ($type == 'forum-topic' || $type == 'all-forum-topics') {
+                return Topic::getMenuTypeInfo($type);
+            }
+        });
+
+        Event::listen('pages.menuitem.resolveItem', function($type, $item, $url, $theme) {
+            if ($type == 'forum-channel' || $type == 'all-forum-channels') {
+                return Channel::resolveMenuItem($item, $url, $theme);
+            } elseif ($type == 'all-forum-topics') {
+                return Topic::resolveMenuItem($item, $url, $theme);
+            }
```

Comment:
Switch statements might be better here, @bennothommo any thoughts?

Reply:
I basically followed the blog implementation here.
https://github.com/wintercms/wn-blog-plugin/blob/main/Plugin.php#L156-L170

Reply:
Ping @bennothommo